### PR TITLE
pkp/pkp-lib#4890 Use public file upload API for image uploads

### PR DESCRIPTION
**Pull request by NateWr**
- Removes justboil.me image upload plugin in favor of
  tinymce's built-in image upload feature
- Fixes custom tinymce params being passed to browser
  error because the old hook fired before plugins
  could hook in

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -16,10 +16,6 @@ License
 This plugin is licensed under the GNU General Public License v2. See the file
 LICENSE for the complete terms of this license.
 
-The jbimages plugin for Tiny is included in this plugin under its own
-license (currently Creative Commons Attribution 3.0 Unported License); see
-justboil.me for details.
```
Comment (asmecher) [inline README:21]: (I AM VERY HAPPY THIS IS GOING -- IT CONTAINED A SUBSTANTIAL AMOUNT OF ILL-MAINTAINED CODEIGNITER MULCH AND WAS GOING TO GET US IN THE BACK AT SOME POINT)

File: TinyMCEPlugin.inc.php [hunk 2]
```diff
@@ -21,7 +21,7 @@ class TinyMCEPlugin extends GenericPlugin {
 	 */
 	function register($category, $path, $mainContextId = null) {
 		if (parent::register($category, $path, $mainContextId)) {
-			if ($this->getEnabled($mainContextId)) {
+			if ($this->getEnabled()) {
```
Comment (asmecher) [inline TinyMCEPlugin.inc.php:24]: Any reason we're no longer providing `$mainContextId` upstream?

Comment (NateWr) [reply TinyMCEPlugin.inc.php:24]: Ah this looks like my mistake. `Plugin::getEnabled()` takes no arguments, but `LazyLoadPlugin::getEnabled()` does.

I've restored the argument there, but should these function signatures be synced?

Comment (asmecher) [reply TinyMCEPlugin.inc.php:24]: No, I think the discrepancy makes sense as it is, at least arguably. The `Plugin` implementation is just a trivial stub; it wouldn't use the parameter at all.

**Pull request merged**